packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx

```diff
@@ -34,19 +34,16 @@ const SceneHierarchyTreeItem: FC<SceneHierarchyTreeItemProps> = ({
 
   const model = getObject3DBySceneNodeRef(key) as Object3D | undefined;
   const [childNodes] = useChildNodes(key);
-  const isValidModelRef = componentTypes?.find(
-    (type) =>
-      type === KnownComponentType.ModelRef &&
-      (type as unknown as IModelRefComponentInternal)?.modelType !== ModelType.Environment,
-  );
-
-  const [{ variation: subModelSelectionEnabled }] = useFeature(COMPOSER_FEATURES[COMPOSER_FEATURES.SubModelSelection]);
-  const showSubModel = subModelSelectionEnabled === 'T1' && isValidModelRef && !!model && !isViewing();
   const sceneComposerId = useContext(sceneComposerIdContext);
   const { getSceneNodeByRef } = useSceneDocument(sceneComposerId);
   const node = getSceneNodeByRef(key);
+  const component = findComponentByType(node, KnownComponentType.ModelRef) as IModelRefComponentInternal;
+  const componentRef = component?.ref;
+  const isValidModelRef = component && component.modelType !== ModelType.Environment;
```

Comment:
useMemo so we don't recalculate every time?

Reply:
Good call out. Fixed.